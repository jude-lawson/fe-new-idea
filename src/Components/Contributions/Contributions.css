.contributions-container {
  background: var(--primary-white);
  margin: 2rem auto;
  max-width: 120rem;
}

.contribution {
  border-bottom: 1px solid grey;
  padding: 2rem;
}

.contribution-title {
  font-size: 2.5rem;
  padding: 2rem;
}

.contribution:first-of-type {
  border-top: 1px solid grey;
}
