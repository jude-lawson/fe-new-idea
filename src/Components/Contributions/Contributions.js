import React from 'react';
import PropTypes from 'prop-types';
import placeholder from '../../images/place-holder.png';
import CommentForm from '../../Containers/CommentForm/CommentForm';
import './Contributions.css';

const Contributions = ({ contributions, hidden, toggleCommentForm }) => {
  const renderContributions = contributions.map(contribution =>
    <div className="contribution" key={contribution.id}>
      <h2 className="idea-title">{contribution.title}</h2>
      <p className="idea-author">
        <span className="idea-image">
          <img
            src={contribution.author.profile_pic_url ?  contribution.author.profile_pic_url : placeholder}
            alt={contribution.author.username}
          />
        </span>
        {contribution.author ? contribution.author.username : null}
      </p>
      <p className="idea-content">{contribution.body}</p>
    </div>
  );

  return (
    <React.Fragment>
      {contributions.length ?
        <section className="contributions-container">
          <div className="contribution-header">
            <h2 className="contribution-title">Contributions</h2>
            <button
              onClick={() => toggleCommentForm()}
              className="btn-grad btn-small blue-btn"
            >
            New Contribution
            </button>
          </div>
          <CommentForm hidden={hidden}/>
          {renderContributions}
        </section> :
        <section className="contributions-container">
          <div className="contribution-header">
            <h2 className="contribution-title">No Contributions Yet</h2>
            <button
              onClick={() => toggleCommentForm()}
              className="btn-grad btn-small blue-btn"
            >
            New Contribution
            </button>
          </div>
          <CommentForm hidden={hidden}/>
        </section>
      }
    </React.Fragment>
  );
};

Contributions.propTypes = {
  contributions: PropTypes.arrayOf(PropTypes.object),
  hidden: PropTypes.bool,
  toggleCommentForm: PropTypes.func
};

export default Contributions;
