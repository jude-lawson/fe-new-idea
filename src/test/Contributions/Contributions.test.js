import React from 'react';
import { shallow } from 'enzyme'
import Contributions from '../../Components/Contributions/Contributions'

describe('<Contributions />', () => {
  let wrapper;
  const mockContributions = [
    {
      id: 1,
      title: 'hello',
      author: {
        username: 'Jeff',
        profile_url_pic: 'SomeURL'
      },
      body: 'world'
    }
  ]
  beforeEach(() => wrapper = shallow(<Contributions contributions={mockContributions} />))

  test('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  });
});
