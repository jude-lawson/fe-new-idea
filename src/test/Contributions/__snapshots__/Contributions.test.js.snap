// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contributions /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contributions
    contributions={
        Array [
            Object {
              "author": Object {
                "profile_url_pic": "SomeURL",
                "username": "Jeff",
              },
              "body": "world",
              "id": 1,
              "title": "hello",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        className="contributions-container"
>
        <div
                className="contribution-header"
        >
                <h2
                        className="contribution-title"
                >
                        Contributions
                </h2>
                <button
                        className="btn-grad btn-small blue-btn"
                        onClick={[Function]}
                >
                        New Contribution
                </button>
        </div>
        <Connect(CommentForm)
                hidden={undefined}
        />
        <div
                className="contribution"
        >
                <h2
                        className="idea-title"
                >
                        hello
                </h2>
                <p
                        className="idea-author"
                >
                        <span
                                className="idea-image"
                        >
                                <img
                                        alt="Jeff"
                                        src="place-holder.png"
                                />
                        </span>
                        Jeff
                </p>
                <p
                        className="idea-content"
                >
                        world
                </p>
        </div>
</section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="contribution-header"
>
            <h2
                        className="contribution-title"
            >
                        Contributions
            </h2>
            <button
                        className="btn-grad btn-small blue-btn"
                        onClick={[Function]}
            >
                        New Contribution
            </button>
</div>,
          <Connect(CommentForm)
            hidden={undefined}
/>,
          Array [
            <div
              className="contribution"
>
              <h2
                            className="idea-title"
              >
                            hello
              </h2>
              <p
                            className="idea-author"
              >
                            <span
                                          className="idea-image"
                            >
                                          <img
                                                        alt="Jeff"
                                                        src="place-holder.png"
                                          />
                            </span>
                            Jeff
              </p>
              <p
                            className="idea-content"
              >
                            world
              </p>
</div>,
          ],
        ],
        "className": "contributions-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="contribution-title"
>
                Contributions
</h2>,
              <button
                className="btn-grad btn-small blue-btn"
                onClick={[Function]}
>
                New Contribution
</button>,
            ],
            "className": "contribution-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contributions",
                "className": "contribution-title",
              },
              "ref": null,
              "rendered": "Contributions",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Contribution",
                "className": "btn-grad btn-small blue-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "New Contribution",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hidden": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="idea-title"
>
                hello
</h2>,
              <p
                className="idea-author"
>
                <span
                                className="idea-image"
                >
                                <img
                                                alt="Jeff"
                                                src="place-holder.png"
                                />
                </span>
                Jeff
</p>,
              <p
                className="idea-content"
>
                world
</p>,
            ],
            "className": "contribution",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hello",
                "className": "idea-title",
              },
              "ref": null,
              "rendered": "hello",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="idea-image"
>
                    <img
                                        alt="Jeff"
                                        src="place-holder.png"
                    />
</span>,
                  "Jeff",
                ],
                "className": "idea-author",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Jeff"
                      src="place-holder.png"
/>,
                    "className": "idea-image",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Jeff",
                      "src": "place-holder.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "span",
                },
                "Jeff",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "world",
                "className": "idea-content",
              },
              "ref": null,
              "rendered": "world",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          className="contributions-container"
>
          <div
                    className="contribution-header"
          >
                    <h2
                              className="contribution-title"
                    >
                              Contributions
                    </h2>
                    <button
                              className="btn-grad btn-small blue-btn"
                              onClick={[Function]}
                    >
                              New Contribution
                    </button>
          </div>
          <Connect(CommentForm)
                    hidden={undefined}
          />
          <div
                    className="contribution"
          >
                    <h2
                              className="idea-title"
                    >
                              hello
                    </h2>
                    <p
                              className="idea-author"
                    >
                              <span
                                        className="idea-image"
                              >
                                        <img
                                                  alt="Jeff"
                                                  src="place-holder.png"
                                        />
                              </span>
                              Jeff
                    </p>
                    <p
                              className="idea-content"
                    >
                              world
                    </p>
          </div>
</section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="contribution-header"
>
              <h2
                            className="contribution-title"
              >
                            Contributions
              </h2>
              <button
                            className="btn-grad btn-small blue-btn"
                            onClick={[Function]}
              >
                            New Contribution
              </button>
</div>,
            <Connect(CommentForm)
              hidden={undefined}
/>,
            Array [
              <div
                className="contribution"
>
                <h2
                                className="idea-title"
                >
                                hello
                </h2>
                <p
                                className="idea-author"
                >
                                <span
                                                className="idea-image"
                                >
                                                <img
                                                                alt="Jeff"
                                                                src="place-holder.png"
                                                />
                                </span>
                                Jeff
                </p>
                <p
                                className="idea-content"
                >
                                world
                </p>
</div>,
            ],
          ],
          "className": "contributions-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="contribution-title"
>
                  Contributions
</h2>,
                <button
                  className="btn-grad btn-small blue-btn"
                  onClick={[Function]}
>
                  New Contribution
</button>,
              ],
              "className": "contribution-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contributions",
                  "className": "contribution-title",
                },
                "ref": null,
                "rendered": "Contributions",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Contribution",
                  "className": "btn-grad btn-small blue-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "New Contribution",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "hidden": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="idea-title"
>
                  hello
</h2>,
                <p
                  className="idea-author"
>
                  <span
                                    className="idea-image"
                  >
                                    <img
                                                      alt="Jeff"
                                                      src="place-holder.png"
                                    />
                  </span>
                  Jeff
</p>,
                <p
                  className="idea-content"
>
                  world
</p>,
              ],
              "className": "contribution",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "hello",
                  "className": "idea-title",
                },
                "ref": null,
                "rendered": "hello",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="idea-image"
>
                      <img
                                            alt="Jeff"
                                            src="place-holder.png"
                      />
</span>,
                    "Jeff",
                  ],
                  "className": "idea-author",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        alt="Jeff"
                        src="place-holder.png"
/>,
                      "className": "idea-image",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "Jeff",
                        "src": "place-holder.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "span",
                  },
                  "Jeff",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "world",
                  "className": "idea-content",
                },
                "ref": null,
                "rendered": "world",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
